src/Microsoft.DotNet.Interactive.Http/HttpKernel.cs

```diff
@@ -124,28 +125,22 @@ async Task IKernelCommandHandler<SendValue>.HandleAsync(SendValue command, Kerne
 
     private Task SetValueAsync(string valueName, object value, Type? declaredType = null)
     {
-        if(value is (HttpVariableDeclarationAndAssignmentNode))
-        {
-            var variable = (HttpVariableDeclarationAndAssignmentNode) value;
-            if(variable.ValueNode is not null)
-            {
-                var bindingResult = variable.ValueNode.TryGetValue(BindExpressionValues);
-                if (bindingResult.IsSuccessful && bindingResult.Value is not null)
-                {
-                    _variables[valueName] = bindingResult.Value;
-                }
-            }
-            
-        } 
-        else
-        {
-            _variables[valueName] = value;
-        }
+        _variables[valueName] = value;
         return Task.CompletedTask;
     }
 
     async Task IKernelCommandHandler<SubmitCode>.HandleAsync(SubmitCode command, KernelInvocationContext context)
     {
+        string? doc = "";
+        if (command.Parameters.TryGetValue("Document", out doc))
```

Comment:
The approach to pass the whole document as a parameter is interesting since that avoids the need to identify which request in the document should be executed.

It does mean that we will be repeating the text for request in the `command` again in the document - but that should hopefully not add too much overhead...

We will however end up reparsing the document and binding all the variables in it before each request, will we not? Or are we being smarter in the http editor and only passing down the document if the document content has changed?

Reply:
I wonder if we should also have a check in the kernel above to only repeat the parsing and binding if the document content has changed. ..

Reply:
Sounds good, I went ahead with the approach to only repeat the parsing if the document has changed